pkg/mcr/registry.go

```diff
@@ -36,23 +36,38 @@ func LoadMachineClassesFile(filename string) ([]iri.MachineClass, error) {
 	return LoadMachineClasses(file)
 }
 
-func NewMachineClassRegistry(classes []iri.MachineClass) (*Mcr, error) {
+func NewMachineClassRegistry(ctx context.Context, classes []iri.MachineClass, disableHugepages bool) (*Mcr, error) {
 	registry := Mcr{
-		classes: map[string]iri.MachineClass{},
+		classes:          map[string]iri.MachineClass{},
+		disableHugepages: disableHugepages,
+	}
+
+	var hugepageSize int64
+	if !disableHugepages {
+		hostMem, err := mem.VirtualMemoryWithContext(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get host memory: %w", err)
```

Comment:
Currently, error messages for `cpu.InfoWithContext` and `mem.VirtualMemoryWithContext` are the same. Could you please change the message for `cpu.InfoWithContext`.